authors/apps/authentication/serializers.py

```diff
@@ -1,28 +1,74 @@
+import re
 from django.contrib.auth import authenticate
-from rest_framework import serializers
+from rest_framework import serializers, exceptions
 from .models import User
 
 
 class RegistrationSerializer(serializers.ModelSerializer):
     """Serializers registration requests and creates a new user."""
 
-    # Ensure passwords are at least 8 characters long, no longer than 128
-    # characters, and can not be read by the client.
     password = serializers.CharField(
-        max_length=128,
-        min_length=8,
         write_only=True
     )
+    """
+    Overide default varidation to make sure users receive descriptive
+    error messages
+    """
+    email = serializers.EmailField()
+    username = serializers.CharField()
 
     # The client should not be able to send a token along with a registration
     # request. Making `token` read-only handles that for us.
 
     class Meta:
         model = User
+
         # List all of the fields that could possibly be included in a request
         # or response, including fields specified explicitly above.
+
         fields = ['email', 'username', 'password']
 
+    def validate_password(self, password):
+        """
+        User's password should be between 8 to 128 characters
+        and  should contain atleas one number and a capital letter
+        """
+        if re.match('^.{8,128}$', password) is None:
```

Comment:
Use `len()` for this. It's cheaper than using regex.

Reply:
Let me work on that